lib/_tls_wrap.js

```diff
@@ -869,9 +869,25 @@ Server.prototype.setOptions = function(options) {
   }
 
   if (options.pfx) this.pfx = options.pfx;
-  if (options.key) this.key = options.key;
+  var defaultCiphers = options.ciphers === tls.DEFAULT_CIPHERS;
+  if (!options.key) {
+    if ((options.ciphers === undefined || defaultCiphers) && !options.pfx) {
+      throw new Error('key is a required parameter for Server.createServer');
+    }
+  } else {
+    this.key = options.key;
+  }
+
   if (options.passphrase) this.passphrase = options.passphrase;
-  if (options.cert) this.cert = options.cert;
+
+  if (!options.cert) {
+    if ((options.ciphers === undefined || defaultCiphers) && !options.pfx) {
+      throw new Error('cert is a required parameter for Server.createServer');
+    }
+  } else {
+    this.cert = options.cert;
+  }
+
   if (options.ca) this.ca = options.ca;
```

Comment:
What happens if I want to set the certificate authority for reading only but not the cert/private key.
Example: http://docs.aws.amazon.com/apigateway/latest/developerguide/getting-started-client-side-ssl-authentication.html